lib/elixir/lib/kernel/typespec.ex

```diff
@@ -962,8 +962,20 @@ defmodule Kernel.Typespec do
 
   defp ensure_no_unused_local_vars!(caller, local_vars) do
     fun = fn
-      {name, :used_once} -> compile_error(caller, "type variable #{name} is unused")
-      _ -> :ok
+      {:_, _used} ->
+        compile_error(caller, "type variable '_' is invalid")
```

Comment:
I would move this check to `update_local_vars`

Reply:
ok, thanks.

Reply:
I noticed ensure_no_unused_local_vars! is also used in translate_spec
it's not works as expected.
I will move this check to new function ensure_no_underscore_local_vars! and use it only in translate_type.